### ISSUE TITLE
Special form `InitVar` not flagged as error when used with `TypeAlias`

### ISSUE THREAD

**Issue by Diogo-Rossi** [microsoft/pyright#7357]
## Environment data

-   Language Server version: 2024.2.2
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.11, Anaconda

## Code Snippet

This works ok
```python
from dataclasses import dataclass, InitVar

@dataclass
class Teste:
    a: str
    b: InitVar[str]

t = Teste('foo','bar')

t.b # ISSUE: Cannot access member "b" for type "Teste"; Member "b" is an init-only field
```
This does not work in the same way
```python
from typing import TypeAlias
from dataclasses import dataclass, InitVar

Alias: TypeAlias = InitVar[str]

@dataclass
class Teste:
    a: str
    b: Alias

t = Teste('foo','bar')

t.b # Pylance does not raise ISSUE here
```

## Repro Steps

1. Use the 2 snippets above in separate python files on VS Code.
2. The first snippet shows an issue when trying to access the init-only field of the dataclass
3. The second snippet does not shows error when trying to access the init-only field of the dataclass
4. The init-only field can be inspected in both situations in run time, using python REPL. It raises an error similar to:
```
----> 1 t.b

AttributeError: 'Teste' object has no attribute 'b'
```

## Expected behavior

I would expect that `InitVar` would be processed as an init-only field  even if it is used from a type alias

## Actual behavior

When using a `TypeAlias`, pylance does not recognize the  init-only field.

**Comment by erictraut** [comment]
This behavior is by design. `InitVar` cannot be used in a type alias because it's not part of a type. It's a qualifier that indicates additional (non-type) metadata about a variable that you declare within a dataclass. Type aliases apply only to type data. You must therefore use `InitVar` directly in the context of the variable declaration if you want it to apply.

**Comment by erictraut** [comment]
I'm going to re-open the bug and rename it because pyright should flag the use of `InitVar` within a type alias as an error.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.352, which I just published. It now properly generates an error if `InitVar` is used in a TypeAlias definition.
